compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1410,6 +1410,33 @@ object SymDenotations {
     def namedType(using Context): NamedType =
       if (isType) typeRef else termRef
 
+    /** The typeRef where `pre.O$.this` is changed to `pre.O.type`
```

Comment:
Perhaps it's worth it to explain this in more details:
```suggestion
    /** Like typeRef, but objects in the prefix are represented by their singleton type,
     *  this means we output `pre.O.member` rather than `pre.O$.this.member`.
```

Reply:
That's better, thanks 👍